### ISSUE TITLE
Site isn't loading - Much errors in console

### ISSUE THREAD

**Issue by sirrobo-development** [ajenti/ajenti#1271]
Hello,

site isn't loading, after installation. I use ubuntu 18.04 with apache2. There a much errors in my browser-console:

```
all.js:2033 Plugin load error:
(anonymous) @ all.js:2033
all.js:2034  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build/all.js
(anonymous) @ all.js:2034
all.js:2035    ReferenceError: angular is not defined
    at all.js:702
(anonymous) @ all.js:2035
all.js:2178 Plugin load error:
(anonymous) @ all.js:2178
all.js:2179  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_datetime/resources/build/all.js
(anonymous) @ all.js:2179
all.js:2180    ReferenceError: angular is not defined
    at all.js:2041
(anonymous) @ all.js:2180
all.js:2353 Plugin load error:
(anonymous) @ all.js:2353
all.js:2354  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_auth_users/resources/build/all.js
(anonymous) @ all.js:2354
all.js:2355    ReferenceError: angular is not defined
    at all.js:2186
(anonymous) @ all.js:2355
all.js:2675 Plugin load error:
(anonymous) @ all.js:2675
all.js:2676  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_dashboard/resources/build/all.js
(anonymous) @ all.js:2676
all.js:2677    ReferenceError: angular is not defined
    at all.js:2361
(anonymous) @ all.js:2677
all.js:2707 Plugin load error:
(anonymous) @ all.js:2707
all.js:2708  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_passwd/resources/build/all.js
(anonymous) @ all.js:2708
all.js:2709    ReferenceError: angular is not defined
    at all.js:2683
(anonymous) @ all.js:2709
all.js:2778 Plugin load error:
(anonymous) @ all.js:2778
all.js:2779  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_ace/resources/build/all.js
(anonymous) @ all.js:2779
all.js:2780    ReferenceError: angular is not defined
    at all.js:2715
(anonymous) @ all.js:2780
all.js:3039 Plugin load error:
(anonymous) @ all.js:3039
all.js:3040  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_filesystem/resources/build/all.js
(anonymous) @ all.js:3040
all.js:3041    ReferenceError: angular is not defined
    at all.js:2786
(anonymous) @ all.js:3041
all.js:3345 Plugin load error:
(anonymous) @ all.js:3345
all.js:3346  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_filemanager/resources/build/all.js
(anonymous) @ all.js:3346
all.js:3347    ReferenceError: angular is not defined
    at all.js:3047
(anonymous) @ all.js:3347
all.js:3481 Plugin load error:
(anonymous) @ all.js:3481
all.js:3482  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_notepad/resources/build/all.js
(anonymous) @ all.js:3482
all.js:3483    ReferenceError: angular is not defined
    at all.js:3353
(anonymous) @ all.js:3483
all.js:3782 Plugin load error:
(anonymous) @ all.js:3782
all.js:3783  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_plugins/resources/build/all.js
(anonymous) @ all.js:3783
all.js:3784    ReferenceError: angular is not defined
    at all.js:3489
(anonymous) @ all.js:3784
all.js:3936 Plugin load error:
(anonymous) @ all.js:3936
all.js:3937  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_settings/resources/build/all.js
(anonymous) @ all.js:3937
all.js:3938    ReferenceError: angular is not defined
    at all.js:3790
(anonymous) @ all.js:3938
all.js:4230 Plugin load error:
(anonymous) @ all.js:4230
all.js:4231  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_augeas/resources/build/all.js
(anonymous) @ all.js:4231
all.js:4232    ReferenceError: angular is not defined
    at all.js:3944
(anonymous) @ all.js:4232
all.js:4843 Plugin load error:
(anonymous) @ all.js:4843
all.js:4844  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_terminal/resources/build/all.js
(anonymous) @ all.js:4844
all.js:4845    ReferenceError: angular is not defined
    at all.js:4238
(anonymous) @ all.js:4845
all.js:5017 Plugin load error:
(anonymous) @ all.js:5017
all.js:5018  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_services/resources/build/all.js
(anonymous) @ all.js:5018
all.js:5019    ReferenceError: angular is not defined
    at all.js:4851
(anonymous) @ all.js:5019
all.js:5190 Plugin load error:
(anonymous) @ all.js:5190
all.js:5191  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_power/resources/build/all.js
(anonymous) @ all.js:5191
all.js:5192    ReferenceError: angular is not defined
    at all.js:5025
(anonymous) @ all.js:5192
all.js:5350 Plugin load error:
(anonymous) @ all.js:5350
all.js:5351  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_packages/resources/build/all.js
(anonymous) @ all.js:5351
all.js:5352    ReferenceError: angular is not defined
    at all.js:5198
(anonymous) @ all.js:5352
all.js:5576 Plugin load error:
(anonymous) @ all.js:5576
all.js:5577  * /usr/local/lib/python3.5/dist-packages/ajenti_plugin_network/resources/build/all.js
(anonymous) @ all.js:5577
all.js:5578    ReferenceError: angular is not defined
    at all.js:5358
(anonymous) @ all.js:5578
all.partials.js:2 Uncaught ReferenceError: angular is not defined
    at all.partials.js:2
(anonymous) @ all.partials.js:2
normal:180 Uncaught ReferenceError: angular is not defined
```

**Comment by kiarn** [comment]
Hello @sirrobo-development 

Can you please post the content of `/var/log/ajenti/ajenti.log` ?

**Comment by domve** [comment]
I have the same issue on Debian 9, the only thing that pops up in the logs is  ```2020-11-10 12:05:43,544 ERROR   : Unhandled endpoint error at /resources/all.vendor.js```

**Comment by kiarn** [comment]
Hello @domve ,

One way to find out the reason is to launch ajenti in dev mode : `/usr/local/bin/ajenti-panel --dev` ( quit with Ctrl + C ). PLease tell me if there's some errors, and I will try to fix it.

**Comment by domve** [comment]
Oh wow, I didn't expect to get a reply this quickly, here is the output if I run it from the terminal:

```
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.004s    0.0 bytes   302 GET /
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /view/login/normal
10.11.2020 12:21  [...    11738]          main.py:55    WARN     Cannot build resources as restricted user nobody
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.003s      10.7 KB   200 GET /view/login/normal
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/all.vendor.css
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.004s       6.3 KB   200 GET /resources/all.vendor.css
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/core/resources/vendor/fontawesome/css/all.min.css
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/all.css
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.012s    0.0 bytes   304 GET /resources/core/resources/vendor/fontawesome/css/all.min.css
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.014s      26.7 KB   200 GET /resources/all.css
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/core/resources/vendor/fontawesome/css/v4-shims.css
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.014s    0.0 bytes   304 GET /resources/core/resources/vendor/fontawesome/css/v4-shims.css
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/core/resources/vendor/pt-sans/styles/pt_sans.css
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.005s    0.0 bytes   304 GET /resources/core/resources/vendor/pt-sans/styles/pt_sans.css
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/all.vendor.js
10.11.2020 12:21  [rstrct 11738]          gate.py:120   ERROR    Unhandled endpoint error at /resources/all.vendor.js
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/aj/api/endpoint.py", line 71, in wrapper
    result = fx(self, context, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/views/resource_server.py", line 41, in handle_build
    file_content = open(path).read()
  File "/usr/lib/python3.5/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 103: ordinal not in range(128)
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/aj/routing.py", line 92, in handle
    output = instance.handle(http_context)
  File "/usr/local/lib/python3.5/dist-packages/aj/api/http.py", line 84, in handle
    data = method(http_context, **http_context.route_data)
  File "/usr/local/lib/python3.5/dist-packages/aj/api/endpoint.py", line 71, in wrapper
    result = fx(self, context, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/views/resource_server.py", line 41, in handle_build
    file_content = open(path).read()
  File "/usr/lib/python3.5/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 103: ordinal not in range(128)
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.008s       6.2 KB   500 GET /resources/all.vendor.js
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/all.init.js
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.002s  252.0 bytes   200 GET /resources/all.init.js
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/all.js
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.009s      30.7 KB   200 GET /resources/all.js
10.11.2020 12:21  [rstrct 11738]          gate.py:120   DEBUG                        ... GET /resources/all.partials.js
10.11.2020 12:21  [master 11731]    middleware.py:250   DEBUG  0.011s      13.9 KB   200 GET /resources/all.partials.js

**Comment by kiarn** [comment]
Ok, maybe it is something wrong with the package on pypi.
Which version of ajenti do you have ? (please check with `python3 -m pip list | grep aj`)
What's the content of the build folder ? 
(please check with `ls /usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build -al`)

**Comment by domve** [comment]
I'm using ajenti 2.1.36, here's the build folder listing:
```
-rwxr-xr-x 1 root staff  196822 Nov 10 11:52 all.css
-rwxr-xr-x 1 root staff   59256 Nov 10 11:52 all.js
-rwxr-xr-x 1 root staff   14580 Nov 10 11:52 all.vendor.css
-rwxr-xr-x 1 root staff 2044538 Nov 10 11:52 all.vendor.js
```
However some other plugins seem to have an empty `all.vendor.js` file, not sure if that's related.

**Comment by kiarn** [comment]
Thanks for the list, I have exactly the same output.

> However some other plugins seem to have an empty `all.vendor.js` file, not sure if that's related.

It's not a problem and a normal behaviour to have some empty `all.vendor.js`.

I tried to make an [automated fresh install](http://docs.ajenti.org/en/latest/man/install.html) on an empty Debian 9.4, and I'm not able to reproduce this bug.
The only difference I see is : `WARN     Cannot build resources as restricted user nobody`. Did you launch ajenti as root ?

**Comment by kiarn** [comment]
The problem is a char © in a javascript file. What is the output of `file /usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build/*` ?

All files should have an utf-8 encoding.

> The only difference I see is : `WARN Cannot build resources as restricted user nobody`. Did you launch ajenti as root ?

This is not related.

**Comment by domve** [comment]
That does seem to be the issue! All of the files were being interpreted as ASCII text. 
```
/usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build/all.css:        ASCII text, with very long lines
/usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build/all.js:         ASCII text, with very long lines
/usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build/all.vendor.css: ASCII text, with very long lines
/usr/local/lib/python3.5/dist-packages/ajenti_plugin_core/resources/build/all.vendor.js:  UTF-8 Unicode text, with very long lines, with LF, NEL line terminators
```

Forcing `open()` to read files as UTF-8 in `ajenti_plugin_core/views/resource_server.py` fixed it, and I was able to get ajenti running.

> Line 41: file_content = open(path, **encoding="UTF-8"**).read()

Also, for the record, I am on Debian 9.13 which is likely why you couldn't reproduce it.

**Comment by kiarn** [comment]
> Forcing `open()` to read files as UTF-8 in `ajenti_plugin_core/views/resource_server.py` fixed it, and I was able to get ajenti running.
> 
> > Line 41: file_content = open(path, **encoding="UTF-8"**).read()

Ok, perfect, thank you, you already answered my next question !
I pushed the fix : https://github.com/kiarn/ajenti/commit/adfd4ac0062032f7d1d20528c24649349697320f

> Also, for the record, I am on Debian 9.13 which is likely why you couldn't reproduce it.

I will give a try on a virtual machine, because I don't really understand why python doesn't use utf-8 encoding as default in this case.

**Comment by nlubello** [comment]
Same issue on Debian distribution on ARM platform.

With the given fix [adfd4ac](https://github.com/kiarn/ajenti/commit/adfd4ac0062032f7d1d20528c24649349697320f) issue was solved
